tardis/opacities/opacity_state.py

```diff
@@ -106,6 +106,41 @@ def __init__(
         self.photo_ion_activation_idx = photo_ion_activation_idx
         self.k_packet_idx = k_packet_idx
 
+    def __getitem__(self, i):
+        """Get a shell or slice of shells of the attributes of the opacity state
+
+        Args:
+            i (int, slice): shell index or slice
+
+        Returns:
+            OpacityState : a shallow copy of the current instance
+        """
+        # NOTE: This currently will not work with continuum processes since it does not slice those arrays
+        return OpacityState(
+            self.electron_density[i],
```

Comment:
check if `i` is an `int` and then raise an error

Reply:
This wouldn't allow one to slice the opacity state then

Reply:
To be clear, Wolfgang is concerned about what happens if somebody requests `OpacityState[5]` and passes that to something else.

Reply:
yep. OpacityState[5] should return either a length 1 OPacity state or fail

Reply:
Which do you prefer?